src/System.Private.CoreLib/shared/System/Buffers/Text/Utf8Formatter/Utf8Formatter.Float.cs

```diff
@@ -89,23 +83,34 @@ public static bool TryFormat(float value, Span<byte> destination, out int bytesW
                 default:
                     return FormattingHelpers.TryFormatThrowFormatException(out bytesWritten);
             }
+            
+            Span<char> formatText = stackalloc char[StandardFormat.FormatStringLength];
+            int formattedLength = format.Format(formatText);
+            formatText = formatText.Slice(0, formattedLength);
 
-            string formatString = format.ToString();
-            string utf16Text = value.ToString(formatString, CultureInfo.InvariantCulture);
-            int length = utf16Text.Length;
-            if (length > destination.Length)
+            // We first try to format into a stack-allocated buffer, and if it succeeds, we can avoid
+            // all allocation.  If that fails, we fall back to allocating strings.  If it proves impactful,
+            // that allocation (as well as roundtripping from byte to char and back to byte) could be avoided by
+            // calling into a refactored Number.FormatSingle/Double directly.
+
+            Span<char> stackBuffer = stackalloc char[128]; // large enough to handle the majority cases
```

Comment:
With a maximum precision of 99, won't this cover all cases?

https://github.com/dotnet/coreclr/blob/80720a220c340e99aa88d04ebae8118eb4fc198b/src/System.Private.CoreLib/shared/System/Number.NumberBuffer.cs#L18

cc @tannergooding

Reply:
No, it won't cover `F99` since that is 99 digits after the decimal point (`double.MaxValue` has some 300 before the decimal point that will also be printed)

Reply:
No, for example double.MaxValue.ToString("F").Length is 312 chars, and double.MaxValue.ToString("F99").Length is 409 chars.

Reply:
Yes. `F` for `float.MaxValue` is 42 characters and `F99` is `139`